internal/datastore/crdb/tx.go

```diff
@@ -106,3 +117,14 @@ func retriable(ctx context.Context, err error) bool {
 	}
 	return pgerr.SQLState() == crdbRetryErrCode
 }
+
+func resetable(ctx context.Context, err error) bool {
```

Comment:
this is a nit, but we could probably use a `sqlState(context, error)` fn that just extracts the error code (that resetable and retriable can use)